main.cpp

```diff
@@ -4607,6 +4607,41 @@ void sign_guts_elf(elf_file* elf, private_t private_key, public_t public_key) {
         new_block.items.push_back(version);
     }
 
+    // Add entry point when signing Arm images
+    std::shared_ptr<image_type_item> image_type = new_block.get_item<image_type_item>();
+    if (settings.seal.sign && image_type != nullptr && image_type->image_type() == type_exe && image_type->cpu() == cpu_arm) {
+        std::shared_ptr<entry_point_item> entry_point = new_block.get_item<entry_point_item>();
+        if (entry_point == nullptr) {
+            std::shared_ptr<vector_table_item> vtor = new_block.get_item<vector_table_item>();
+            uint32_t vtor_loc = 0x10000000;
+            if (elf->header().entry >= SRAM_START) {
+                vtor_loc = 0x20000000;
+            } else if (elf->header().entry >= XIP_SRAM_START_RP2350) {
+                vtor_loc = 0x13ffc000;
+            } else {
+                vtor_loc = 0x10000000;
+            }
+            if (vtor != nullptr) {
+                vtor_loc = vtor->addr;
+            } else {
+                std::shared_ptr<rolling_window_delta_item> rwd = new_block.get_item<rolling_window_delta_item>();
```

Comment:
note arguably rolling window delta only applies to flash binaries; silly to have it in another binary, but still we could move that into flash oly